### PR TITLE
Feat: Help page

### PR DESCRIPTION
**Pull request by samuelroland**
image
This PR adds a nice help page accessible via `?`.

It also adds methods `Key.describe()` to get description, `Key.name()` to get the case name or a hardcoded symbol (Interrogation -> ?), and `Key.alt()` to indicate a possible alternative.

It needs a new crate `strum` to be able to iterate on enum cases and get the enum case name as string.

We couldn't have a completely dynamic system, this structure still need manual maintenance of alternatives as they are only defined in the UI.

### CODE REVIEW

File: src/core/app/event_handler.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ impl App {
             Key::P => todo!(),
             Key::E => todo!(),
             Key::Esc => self.on_esc(),
+            Key::Interrogation => self.go_to_help(Box::new(self.ui_state.clone()), 0),
```
Comment (AndreCostaaa) [inline src/core/app/event_handler.rs:21]: Here we need to check first if we're not already in the help page else we won't be able to exit
To follow the rest of the code base please add a new fn "on_question_mark" that checks the current state and only changes if we're not already there.
We should also put a big comment at the top of the function so we don't fall on this trap

File: src/models/key.rs [hunk 2]
```diff
@@ -11,4 +13,46 @@ pub enum Key {
...
+        match self {
+            Key::Interrogation => "?".to_string(),
+            _ => format!("{:?}", self).to_lowercase(),
+        }
+    }
+
+    /// An alternative key, can be empty
+    pub fn alt(&self) -> &str {
+        match self {
+            Key::Q => "ctrl+c",
```
Comment (AndreCostaaa) [inline src/models/key.rs:50]: If we want to support things like Ctrl+ C we need to map the sigterm and sigint signals and properly quit the applicaton else we risk leaving the terminal in a weird state as the teardown function won't get called

**Review by AndreCostaaa — CHANGES_REQUESTED** [review]
The page looks good for now but there's one bug embedded in this PR aswell as a feature request in the core part

I'll let someone else review the Ui part

File: src/core/app/event_handler.rs (hunk 1 above)
Comment (G-LK) [reply src/core/app/event_handler.rs]: function added, need to check if it's correct 

https://github.com/plx-pdg/plx/pull/89/commits/667d9c8ef8d7552ebf63893740ea8b6302b07147

File: src/models/key.rs (hunk 2 above)
Comment (G-LK) [reply src/models/key.rs:50]: Where should we implement this ? We supposed it must be integrated by the core (app) ?

Comment (AndreCostaaa) [reply src/models/key.rs:50]: For now we can just remove this line so the user doesn't think we support `ctrl + c`
We can create an issue and add that feature later. It's not a trivial task

File: src/core/app/event_handler.rs [hunk 3]
```diff
@@ -18,9 +18,22 @@ impl App {
...
         }
     }
+
+    // Function to check if we're not already on the help page else we won't be able to exit.
+    pub(super) fn is_on_help(&mut self, last_state: Box<UiState>, scroll_offset: usize) {
+        if self.ui_state
+            != (UiState::Help {
+                last_state,
+                scroll_offset,
+            })
```
Comment (AndreCostaaa) [inline src/core/app/event_handler.rs:31]: Please check [matches](https://doc.rust-lang.org/std/macro.matches.html) to test that the enum variant is the same without checking the inner values. An example: https://github.com/plx-pdg/plx/blob/5fcd78d85b28490bb84e29af55e4c0ea4a500833/src/core/check/output_checker.rs#L84

File: src/core/app/event_handler.rs [hunk 4]
```diff
@@ -18,9 +18,22 @@ impl App {
             Key::P => todo!(),
             Key::E => todo!(),
             Key::Esc => self.on_esc(),
-            Key::Interrogation => self.go_to_help(Box::new(self.ui_state.clone()), 0),
+            Key::Interrogation => self.is_on_help(Box::new(self.ui_state.clone()), 0),
```
Comment (AndreCostaaa) [inline src/core/app/event_handler.rs:21]: Would you be okay renaming this to `self.on_interrogation` to better follow the convention used with the other keys ?

**Review by AndreCostaaa — CHANGES_REQUESTED** [review]

File: src/models/key.rs (hunk 2 above)
Comment (AndreCostaaa) [reply src/models/key.rs:50]: But yes, in app would be best but maybe not doable, main would be another possibility

File: src/core/app/event_handler.rs (hunk 4 above)
Comment (G-LK) [reply src/core/app/event_handler.rs]: yes function renamed

File: src/core/app/event_handler.rs (hunk 3 above)
Comment (G-LK) [reply src/core/app/event_handler.rs]: use matches for the on_interrogation function :

https://github.com/plx-pdg/plx/pull/89/commits/a530123afaf4c38a6dcac48fef9c1a39009cac9e

File: src/models/key.rs (hunk 2 above)
Comment (G-LK) [reply src/models/key.rs]: Keybind commented for now

35e758e449d35d4d65b148bcdf35ba36c5882dee

**Pull request merged**